### PR TITLE
fix(ext/web): webstorage has trap for symbol

### PR DESCRIPTION
**Pull request by petamoriken**
Fixed a bug that webstorage `has` trap that passed `symbol` to `storage.getItem`.

```
> Symbol() in localStorage
Uncaught TypeError: Failed to execute 'getItem' on 'Storage': Argument 1 is a symbol, which cannot be converted to a string
    at makeException (ext:deno_webidl/00_webidl.js:88:10)
    at Array.converters.DOMString (ext:deno_webidl/00_webidl.js:392:11)
    at Storage.getItem (ext:deno_webstorage/01_webstorage.js:58:29)
    at Object.has (ext:deno_webstorage/01_webstorage.js:122:24)
    at <anonymous>:2:10
```

### CODE REVIEW

File: ext/webstorage/01_webstorage.js [hunk 1]
```diff
@@ -83,51 +83,54 @@ function createStorage(persistent) {
...
+        return ReflectGet(target, key, receiver);
       }
+      return target.getItem(key) ?? undefined;
     },
+
     set(target, key, value) {
-      if (typeof key == "symbol") {
-        ObjectDefineProperty(target, key, {
+      if (typeof key === "symbol") {
+        return ReflectDefineProperty(target, key, {
```
Comment (petamoriken) [inline ext/webstorage/01_webstorage.js:113]: `defineProperty` is used here, should it be `Reflect.set`?

**Review by hexiaokang — APPROVED** [review]

**Comment by bartlomieju** [comment]
@crowlKats please take a look

File: ext/webstorage/01_webstorage.js (hunk 1 above)
Comment (crowlKats) [reply ext/webstorage/01_webstorage.js:113]: if my memory serves me right, there were some very weird behavioural differences that caused issues, but I might be wrong.

**Review by crowlKats — APPROVED** [review]
LGTM, thanks

**Pull request merged**